### ISSUE TITLE
sync and threading clients raise error on Windows 7

### ISSUE THREAD

**Issue by vstadnytskyi** [caproto/caproto#540]
I just get an error when I am trying to use Threading client on Windows 7 machine. I do have the latest caproto available which is 0.4.1

```
pip3 install --upgrade caproto
Requirement already up-to-date: caproto in c:\users\user\appdata\local\prog
rams\python\python37\lib\site-packages (0.4.1)
```

My python code:
```
Python 3.7.4 (tags/v3.7.4:e09359112e, Jul  8 2019, 20:34:20) [MSC v.1916 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license()" for more information.
>>> import caproto
>>> caproto.__version__
'0.4.1'
>>> from caproto.threading.client import Context
>>> ctx = Context()
Traceback (most recent call last):
  File "<pyshell#3>", line 1, in <module>
    ctx = Context()
  File "C:\Users\femtoland\AppData\Local\Programs\Python\Python37\lib\site-packages\caproto\threading\client.py", line 936, in __init__
    broadcaster = SharedBroadcaster()
  File "C:\Users\femtoland\AppData\Local\Programs\Python\Python37\lib\site-packages\caproto\threading\client.py", line 342, in __init__
    self.broadcaster.our_address = self.udp_sock.getsockname()[:2]
OSError: [WinError 10022] An invalid argument was supplied
>>> 
```

**Comment by vstadnytskyi** [comment]
There is something funny with Windows 7. The sync client doesn't work either. But It works on mac.

Windows:
```
Python 3.7.4 (tags/v3.7.4:e09359112e, Jul  8 2019, 20:34:20) [MSC v.1916 64 bit (AMD64)] on win32
Type "help", "copyright", "credits" or "license()" for more information.
>>> import caproto
>>> caproto.__version__
'0.4.1'
>>> from caproto.sync.client import read
>>> read('NIH:TEMP.RBV')
Traceback (most recent call last):
  File "<pyshell#5>", line 1, in <module>
    read('NIH:TEMP.RBV')
  File "C:\Users\femtoland\AppData\Local\Programs\Python\Python37\lib\site-packages\caproto\sync\client.py", line 256, in read
    chan = make_channel(pv_name, udp_sock, priority, timeout)
  File "C:\Users\femtoland\AppData\Local\Programs\Python\Python37\lib\site-packages\caproto\sync\client.py", line 133, in make_channel
    address = search(pv_name, udp_sock, timeout)
  File "C:\Users\femtoland\AppData\Local\Programs\Python\Python37\lib\site-packages\caproto\sync\client.py", line 51, in search
    b.our_address = udp_sock.getsockname()[:2]
OSError: [WinError 10022] An invalid argument was supplied
```
Mac:
```
$ ipython3
Python 3.7.4 (v3.7.4:e09359112e, Jul  8 2019, 14:54:52) 
Type 'copyright', 'credits' or 'license' for more information
IPython 7.8.0 -- An enhanced Interactive Python. Type '?' for help.

In [1]: import caproto                                                                                                                                           

In [2]: caproto.__version__                                                                                                                                      
Out[2]: '0.4.1'

In [3]: from caproto.sync.client import read                                                                                                                     

In [4]: read('NIH:TEMP.RBV')                                                                                                                                     
Out[4]: ReadNotifyResponse(data=array([24.0013752]), data_type=<ChannelType.DOUBLE: 6>, data_count=1, status=CAStatusCode(name='ECA_NORMAL', code=0, code_with_severity=1, severity=<CASeverity.SUCCESS: 1>, success=1, defunct=False, description='Normal successful completion'), ioid=0, metadata=None)

In [5]:    
```

**Comment by vstadnytskyi** [comment]
This is related to #514

**Comment by vstadnytskyi** [comment]
I have downgraded caproto to 0.3.4 on my windows machines and the majority of the code (except caproto.logging) is working fine.

**Comment by danielballan** [comment]
Good, I'm glad that downgrading is a viable workaround for now. Will try to get a fix out for this in the next couple weeks. FWIW, I just released v0.4.2 which includes #529.

**Comment by danielballan** [comment]
Closed by #552, released in v0.4.3 in January.
